### DISCUSSION TITLE
Synapse Dedicated: DW_COMPATIBILITY_LEVEL & MCD hash distribution

### DISCUSSION THREAD

**Discussion by brain246** [microsoft/DacFx#220] [Q&A] [answered]
Hi there!

Not sure if this is the right place to ask/discuss this, but i am currently developing a Synapse dedicated pool database with VS2022/Data-Tools/SQl-Database-Project. 

Since we had performance issues with some tables, i set database scoped configuration setting "DW_COMPATIBILITY_LEVEL" to 9000. That enables (since August 2022) the multi column hash distribution feature, meaning since half a year this feature is already in public preview. In our project i cannot live without it anymore 😄 

[Official announcement](https://techcommunity.microsoft.com/t5/azure-synapse-analytics-blog/azure-synapse-analytics-august-update-2022/ba-p/3535126#TOCREF_1)

Now that setting is neither available in the VS project (as database setting), nor can i set it in a (build) script myself as this leads to build & intellisense errors.

When i do a schema compare & update from dev-database to the project, the multi column distribution code is written correctly to the local project files. However this results, again, in intellisense and build errors.

Is my assumption correct, that you will deal with this, as soon as it is out of public preview and becomes a regular feature? Or is there anything i am missing and it should work already? Maybe anyone knows a workaround? 🤣 

Have a nice weekend,
Thomas

**Comment by dzsquared** [comment, marked as answer]
Thanks for raising this! I've opened an issue for it and in good news this was on the team's radar.

**Reply by brain246** [reply]
@dzsquared: thank you! looking forward 👍🏼
